src/circuit/gadget/utilities/enable_flag.rs

```diff
@@ -0,0 +1,230 @@
+use super::{copy, CellValue, UtilitiesInstructions};
+use halo2::{
+    circuit::{Chip, Layouter},
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Permutation, Selector},
+    poly::Rotation,
+};
+use pasta_curves::arithmetic::FieldExt;
+use std::marker::PhantomData;
+
+pub trait EnableFlagInstructions<F: FieldExt>: UtilitiesInstructions<F> {
+    /// Given a `value` and an `enable_flag`, check that either `value = 0`
+    /// or `enable_flag = 1`.
```

Comment:
This is a minor nit that is nonblocking. I prefer "enforce that `value` be zero or, if nonzero, that `enable_flag` must be `1`". (It can be the case that `value = 0` and `enable_flag = 1`, the "either x or y" here is potentially misleading.)